src/imitation/util/rollout.py

```diff
@@ -16,18 +16,44 @@
 
 from . import util  # Relative import needed to prevent cycle with __init__.py
 
-TrajectoryList = List[Dict[str, np.ndarray]]
-"""A list of trajectory dicts.
 
-Each dict contains the keys 'act', 'obs', and 'rew'. For details on these
-key-value pairs, see the docstring for `generate_trajectories`.
-"""
+class Trajectory(NamedTuple):
```

Comment:
I think it's easy to mix up the two NamedTuples since they have similar fields and class names.

Renaming `Trajectory` => `Demonstration` might be nice.
This would necessitate a lot of downstream manual renaming of local variables and methods though, so probably best to make to punt this to an issue.

Reply:
I find `Demonstration` more confusing. Both `Trajectory` and `Transitions` can hold demonstrations. Some algorithms are trajectory-centric (e.g. GCL, deep RL from human preferences), others are fine with unordered transitions (e.g. most IRL algorithms, BC).

I'd be open to renaming these in general though.

Reply:
Ok, given that explanation these names seem fine!